extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1304,23 +1303,67 @@ export class Bind {
 
   /**
    * Print out the current state in the console.
-   * @param {string=} opt_expression
+   * @param {(!Element|string)=} opt_elementOrExpr
    * @private
    */
-  printState_(opt_expression) {
-    if (opt_expression) {
-      if (typeof opt_expression !== 'string') {
-        user().error(TAG, 'Invalid JSON expression. Example: ' +
-            'AMP.printState("foo.bar") to print current value of "foo.bar".');
-        return;
+  debugPrintState_(opt_elementOrExpr) {
+    if (opt_elementOrExpr) {
+      if (typeof opt_elementOrExpr === 'string') {
+        const value = getValueForExpr(this.state_, opt_elementOrExpr);
+        user().info(TAG, value);
+      } else if (opt_elementOrExpr.nodeType === Node.ELEMENT_NODE) {
+        const element = user().assertElement(opt_elementOrExpr);
+        this.debugPrintElement_(element);
+      } else {
+        user().info(TAG, 'Invalid argument. Pass a JSON expression or an ' +
+            'element instead e.g. AMP.printState("foo.bar") or ' +
+            'AMP.printState($0) after selecting an element.');
       }
-      const value = getValueForExpr(this.state_, opt_expression);
-      user().info(TAG, value);
     } else {
       user().info(TAG, this.state_);
     }
   }
 
+  /**
+   * Print out the element's bound attributes and respective expression values.
+   * @param {!Element} element
+   * @private
+   */
+  debugPrintElement_(element) {
+    const index = findIndex(this.boundElements_, boundElement => {
+      return boundElement.element === element;
+    });
+    if (index < 0) {
+      user().info(TAG, 'Element has no bindings:', element);
+      return;
+    }
+    // Evaluate expressions in bindings in `element`.
+    const promises = [];
+    const {boundProperties} = this.boundElements_[index];
+    boundProperties.forEach(boundProperty => {
+      const {expressionString} = boundProperty;
+      promises.push(this.evaluateExpression_(expressionString, this.state_));
+    });
+    // Print the map of attribute to expression value for `element`.
+    const output = map();
```

Comment:
place the constant 'output' inside the 'then'

`Promise.all(promises).then(results => {
      const output = map();
      boundProperties.forEach((boundProperty, i) => {
        const {property} = boundProperty;
...
`

Reply:
Done.